microceph/ceph/osd.go

```diff
@@ -463,6 +520,17 @@ func (m *OSDManager) bootstrapOSD(osdDataPath string, nr int64, wal, db *types.D
 			return fmt.Errorf("failed to set stable path for WAL: %w", err)
 		}
 
+		// Check for partitions on WAL device unless wipe is enabled
+		if !wal.Wipe {
+			hasPartitions, err := m.checkDeviceHasPartitions(storage, wal.Path)
+			if err != nil {
+				return fmt.Errorf("failed to check partitions on WAL device %s: %w", wal.Path, err)
+			}
+			if hasPartitions {
+				return fmt.Errorf("WAL device %s has partitions - use --wipe to override", wal.Path)
+			}
```

Comment:
Why not make this a function that takes a `types.DiskParameter` instead of doing this for the data, WAL and DB devices separatedly? Something like the following:

```
func checkPartitions (disk *types.DiskParameter, m *OSDManager, storage *api.ResourcesStorage) (bool, error) {
  if disk.Wipe {
    hasPartitions, err := m.checkDeviceHasPartitions(storage, disk.Path)
    if err != nil {
      return fmt.Errorf("failed to check partitions on device %s: %w", disk.Path, err)
    } else if hasPartitions {
      return fmt.Errorf("device %s has partitions", disk.Path)
    }
  }
}
```